support/k8s/k8s-workload-registrar/webhookhandler.go

```diff
@@ -38,25 +117,78 @@ func (h *WebhookHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	var in admv1beta1.AdmissionReview
-	if err := json.NewDecoder(req.Body).Decode(&in); err != nil {
-		http.Error(w, "Malformed JSON body", http.StatusBadRequest)
+	var body []byte
+	if req.Body != nil {
+		if data, err := ioutil.ReadAll(req.Body); err == nil {
+			body = data
+		}
+	}
+
+	obj, gvk, err := deserializer.Decode(body, nil, nil)
+	if err != nil {
+		msg := fmt.Sprintf("Malformed JSON body: %v", err)
+		klog.Error(msg)
+		http.Error(w, msg, http.StatusBadRequest)
 		return
 	}
 
-	if in.Request == nil {
-		http.Error(w, "AdmissionReview is missing request", http.StatusBadRequest)
+	admit := newDelegateToV1AdmitHandler(h.controller.ReviewAdmission)
+	ctx := context.TODO()
```

Comment:
We should probably do a 
```
ctx, cancel := context.WithCancel(context.Background())
defer cancel()
```
at the top of the function.

Reply:
The `http.Request` has a [context](https://pkg.go.dev/net/http#Request.Context) available that will be cancelled if the transport dies.

Reply:
thank you folks.